interfaces/builtin/dummy.go

```diff
@@ -0,0 +1,99 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/snap"
+)
+
+type testSnapdInterface struct{}
+
+const testSnapdInterfaceSummary = `snapd dummy test interface`
+const testSnapdInterfaceBaseDeclarationSlots = `
+  dummy:
+    allow-installation:
+      slot-snap-type:
+        - app
+    deny-auto-connection: true
+    deny-connection:
+      on-classic: false
+`
+
+func (iface *testSnapdInterface) String() string {
+	return iface.Name()
+}
+
+// Name returns the name of the dummy interface.
+func (iface *testSnapdInterface) Name() string {
+	return "dummy"
+}
+
+func (iface *testSnapdInterface) StaticInfo() interfaces.StaticInfo {
```

Comment:
Type name needs to match the interface name as usual.

Reply:
Done.